mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/utils/DeleteByQueryExecutor.java

```diff
@@ -57,15 +55,15 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(DeleteByQueryExecutor.class);
 
     private final DeletedMessageVault deletedMessageVault;
-    private final Supplier<Publisher<User>> userWithVaults;
+    private final Publisher<User> userWithVaults;
 
-    public DeleteByQueryExecutor(DeletedMessageVault deletedMessageVault, Supplier<Publisher<User>> userWithVaults) {
+    public DeleteByQueryExecutor(DeletedMessageVault deletedMessageVault, Publisher<User> userWithVaults) {
```

Comment:
The fact that you expect the Flux to republish its values each time you ask is implicit and thus dangerous in my opinion. You can pass a not-replaying Publisher and the code will fail.
You could define a interface with this method in it (but it's basically what you had with the Supplier version)

Reply:
I gonna revert to the supplier version then, aggree, at least it is more explicit.